improver/calibration/ensemble_calibration.py

```diff
@@ -884,6 +1080,113 @@ def mask_cube(cube, landsea_mask):
         else:
             cube.data = np.ma.masked_invalid(cube.data)
 
+    def guess_and_minimise(
+        self,
+        truths,
+        historic_forecasts,
+        forecast_predictor,
+        forecast_var,
+        number_of_realizations,
+    ):
+        """Function to consolidate calls to compute the initial guess, compute
+        the optimised coefficients using minimisation and store the resulting
+        coefficients within a CubeList.
+
+        Args:
+            truths (iris.cube.Cube):
+                Truths from the training dataset.
+            historic_forecasts (iris.cube.Cube):
+                Historic forecasts from the training dataset.
+            forecast_predictor (iris.cube.Cube):
+                Predictor of the forecast within the minimisation. This
+                is either the ensemble mean or the ensemble realizations.
+            forecast_var (iris.cube.Cube):
+                Variance of the forecast for use in the minimisation.
+            number_of_realizations (int or None):
+                Number of realizations within the forecast predictor. If no
+                realizations are present, this option is None.
+
+        Returns:
+            iris.cube.CubeList:
+                CubeList constructed using the coefficients provided and using
+                metadata from the historic_forecasts cube. Each cube within the
+                cubelist is for a separate EMOS coefficient e.g. alpha, beta,
+                gamma, delta.
+
+        """
+        if self.each_point:
+            # Disable ordinary least squares fit when computing each point
+            # independently due to potential for too few data points to create
+            # a reliable fit.
```

Comment:
I've modified this now by enabling the use of the `each_point` option with `realizations`, so that an initial guess is computed separately for each point using linear regression, if these options are chosen. This was already supported but I wasn't quite confident about the behaviour but, I think, it seems fine.